bindings/wallet-core/src/error.rs

```diff
@@ -86,6 +89,12 @@ pub enum ErrorKind {
 
     /// authentication failed
     SymmetricCipherInvalidPassword,
+
+    /// invalid bech32
+    InvalidBech32Hrp {
+        expected: &'static str,
```

Comment:
Unless you really want to have the common error for other similar cases, the error variant could be more specific: `InvalidVoteEncryptionKey`, then you don't need the `expected` pointer.

Reply:
Could also be a sub-cause of `InvalidInput` like the error about invalid key bytes.

Reply:
Yeah, I guess this is a bit overkill, it's partly because I started out with a bit of a different idea. I'll reduce it to `InvalidInput` with details. There is a problem with the fact that there is currently no way of recovering those details from cordova, but that should probably be tackled somewhere else.

Reply:
Alternatively, you could ramify both error cases into specific `Error` variants that are exposed in a non-stringized way through cordova, there just should be some consistency.

Reply:
I think a single `InvalidVoteEncryptionKey` variant should do for both cases, we can add a different error type as `details` to make it more granular, but as I said that's not exposed to cordova right now, and I'd prefer to do it separately.

I don't really understand the part about `non-stringized` way though, as that's what `Error` in javascript is, even if I don't like it (?)

One thing we could do, is map each variant to a different javascript type with `Error` as prototype, I guess, but that requires keeping both things in sync.